oxide-vpc/src/engine/icmpv6.rs

```diff
@@ -155,24 +159,74 @@ fn add_icmpv6_rules(
     // And then pass through unchanged any ICMPv6 Echo Request / Reply messages.
     // These will not be matched by the ping rules above, meaning they're
     // destined for somewhere else. We can add more exceptions to the Deny above
-    // as needed. Note that we don't match on the IP addresses, expecting those
-    // to be handled at the router layer.
+    // as needed.
     let priority = u16::try_from(n_actions + 1).unwrap();
+
+    // Outbound echo messages.
     let mut rule = Rule::new(
         priority,
-        Action::Static(Arc::new(Identity::new("allow-icmpv6-echo"))),
+        Action::Static(Arc::new(Identity::new("allow-outbound-icmpv6-echo"))),
     );
+    // IPv6 only.
+    rule.add_predicate(Predicate::InnerEtherType(vec![EtherTypeMatch::Exact(
+        ETHER_TYPE_IPV6,
+    )]));
+    // From client MAC.
+    rule.add_predicate(Predicate::InnerEtherSrc(vec![EtherAddrMatch::Exact(
+        cfg.private_mac,
+    )]));
+    // To OPTE MAC.
+    rule.add_predicate(Predicate::InnerEtherDst(vec![EtherAddrMatch::Exact(
+        cfg.gateway_mac,
+    )]));
+    // From client IPs.
+    rule.add_predicate(Predicate::InnerSrcIp6(vec![
+        Ipv6AddrMatch::Exact(src_ips[0]),
+        Ipv6AddrMatch::Exact(src_ips[1]),
```

Comment:
I don't think an ICMP echo going to another guest or external network can use the link-local address.

Reply:
Oh duh, will remove.